### DISCUSSION TITLE
local TESS lightcurves

### DISCUSSION THREAD

**Discussion by ghost** [lightkurve/lightkurve#1110] [Q&A] [answered]
Hi all,
I have downloaded in a machine all the lightcurves of SC data obtained in year 1 and 2 of TESS.
I am wondering if there is a method to open all the relevant fits files for a particular TICID.
If the TICID is observed only once, there is no problem (60% of the cases) because I am using the read method. 
However I not able to find a solution for the TICID observed many times.
There is some method that I am overlooking or some workaround that is it known ?

cheers

**Comment by barentsen** [comment, marked as answer]
This is an excellent question @barbierimauro.  Thank you for asking! :+1:

The only way Lightkurve currently supports loading all the light curves for a single TICID is via the archive search feature. For example, to obtain a `LightCurveCollection` which contains all the 2-minute light curves for "TIC 261136679" produced by the official SPOC pipeline, you can use:
```python
collection = lightkurve.search_lightcurve("TIC 261136679", author="SPOC", exptime=120).download_all()
```

This will not use the FITS files you already downloaded.  Fortunately, it will cache all files in the cache directory (`~/.lightkurve-cache`), so at least it won't download the same light curve more than once.   It is possible to change the cache location, e.g. using `download_all(download_dir="/your/directory")`, but any light curves already located in `/your/directory` would only be re-used if they followed the exact directory layout expected by the caching mechanism, which usually isn't practical.

A better solution to use the bulk-downloaded files on your disk may be to use Python's built-in `glob` module to automatically discover the light curve file names that belong to a single TICID.  For example, something like this might work very well for you:
```python
import glob
import lightkurve as lk

ticid = 261136679
lightcurves = []
for filename in glob.glob(f"location/of/your/light/curves/*{ticid}*/*lc.fits"):
    lightcurves.append(lk.read(filename))
collection = lk.LightCurveCollection(lightcurves)
```

Alternatively, if you think Lightkurve should better support searching/loading light curves from your local disk, the best way to make this happen is to come up with an example of what the feature should look like (i.e., feel free to post a piece of pseudocode which doesn't work yet, but which you wish would work!)

Does this help?  I'd be curious to learn what you end up doing, so please report back!

**Reply by ghost** [reply]
@barentsen thanks for the detailed answer, it was very helpful. I was supposing that the solution was based on Collection but I was not able to figure out how to implement, and it was much easier than I was thinking.
After I will implement and test it, I tell you how is working.
